### PR TITLE
Docker health check

### PR DESCRIPTION
**Pull request by quic-amitraj**
PR is not ready to review. I will post in group as soon as its ready.

### CODE REVIEW

File: scripts/Jenkinsfile [hunk 1]
```diff
@@ -1,43 +1,57 @@
...
+        CONTAINER_NAME = "qeff_container"
+    }
     stages 
     {
+        stage('Docker image build')
+        {
+            steps
+            {
+                sh '''
+                    python3 /home/ubuntu/amitraj/docker_changec/qaic-docker/build_image.py --image_name ubuntu22_nightly --user_specification_file /home/ubuntu/amitraj/docker_changec/qaic-docker/sample_user_specs/user_image_spec_qaic.json --log_level 1 --apps_sdk /home/ubuntu/sdks/qaic-apps-1.17.0.8.zip --platform_sdk /home/ubuntu/sdks/qaic-platform-sdk-x86_64-deb-1.17.0.8.zip
```
Comment (ochougul) [inline scripts/Jenkinsfile:21]: Don't use static file, pull it from gerrit

**Comment by ochougul** [comment]
Let's pull nightly latest docker, for testing.

File: scripts/Jenkinsfile [hunk 2]
```diff
@@ -1,43 +1,58 @@
 pipeline
```
Comment (irajagop) [inline scripts/Jenkinsfile:1]: Can we make this a separate CI pipeline? that runs only when `pyproject.toml` is touched upon, or runs only on release branches. The current pipeline by itself takes a lot of time to run, adding this for every PR and every push is not a good idea.

**Review by irajagop — CHANGES_REQUESTED** [review]

File: scripts/Jenkinsfile (hunk 2 above)
Comment (ochougul) [reply scripts/Jenkinsfile:1]: We will be pulling nightly docker which will pull only once every day. Let's wait for the CI time with this pipeline to make a decision on this.

**Comment by vbaddi** [comment]
@quic-amitraj can you add a good summary of what exact docker tests are being targeted from this PR? and how it will reflect in CI pipeline.

File: scripts/Jenkinsfile (hunk 2 above)
Comment (vbaddi) [reply scripts/Jenkinsfile:1]: Agree, docker essentially checks for creation of venv only, and does basic check of installation success, something like `python -c "import QEfficient"`, we should keep it specific to changes in packages only for now. (i.e trigger only when packages are changed)

**Review by vbaddi — CHANGES_REQUESTED** [review]

**Comment by ochougul** [comment]
Let's fix the tests to use the same cache dir.

File: scripts/Jenkinsfile (hunk 2 above)
Comment (quic-amitraj) [reply scripts/Jenkinsfile:1]: Update- 
1. I've successfully implemented a nightly build pipeline that retrieves the latest Docker image from the registry at midnight. This ensures that we're always working with the most up-to-date image.

2. Leveraging this setup, we can seamlessly run tests throughout the day by creating Docker containers. 

3. Additionally, any modifications to the "pyproject.toml" file will trigger a new pipeline. This pipeline will meticulously build the Docker image from scratch and execute tests within it. Our testing process remains robust and aligned with the latest code changes. (Under progress)

File: tests/cloud/test_infer.py [hunk 3]
```diff
@@ -47,6 +47,7 @@ def test_infer(setup, mocker):
         mxfp6=ms.mxfp6,
         mxint8=ms.mxint8,
         device_group=ms.device_group,
+        cache_dir="/root/.cache/huggingface/hub",
```
Comment (ochougul) [inline tests/cloud/test_infer.py:50]: Please use this from environment variable. If set use that else use `None`.

File: scripts/Jenkinsfile [hunk 4]
```diff
@@ -1,43 +1,48 @@
...
         {
             steps
             {
                 sh '''
-                    python3.8 -m venv preflight_qeff
-                    . preflight_qeff/bin/activate
-                    pip install --upgrade pip setuptools
-                    pip install .[test]
-                    rm -rf QEfficient
+                    docker run --privileged -dit --name ${CONTAINER_NAME} -v ./:/data -v  /home/ubuntu/.cache/huggingface/hub:/root/.cache/huggingface/hub docker-registry.qualcomm.com/qraniumtest/qranium:master_latest
```
Comment (ochougul) [inline scripts/Jenkinsfile:22]: Let's use ENV variables for mounting directories, do not expose directories here.

**Review by ochougul — CHANGES_REQUESTED** [review]

File: tests/cloud/test_infer.py [hunk 5]
```diff
@@ -47,6 +47,7 @@ def test_infer(setup, mocker):
         mxfp6=ms.mxfp6,
         mxint8=ms.mxint8,
         device_group=ms.device_group,
+        cache_dir=os.environ.get("docker_huggingface_hub_path"),
```
Comment (ochougul) [inline tests/cloud/test_infer.py:50]: Please make sure all environment variables are in all caps. keep it consistent

File: scripts/Jenkinsfile [hunk 6]
```diff
@@ -1,44 +1,55 @@
...
-                    rm -rf QEfficient
-                '''
+
+                script
+                {
+                    sh '''
+                    . ~/.bashrc
+                    docker run --privileged -dit --name ${CONTAINER_NAME} -v ./:/data -v  ${huggingface_hub_path}:${docker_huggingface_hub_path} ${docker_latest}:master_latest
+                    docker exec ${CONTAINER_NAME} bash -c "
+                    cd /data &&
```
Comment (ochougul) [inline scripts/Jenkinsfile:28]: why `/data` shouldn't it be `efficient_transformers` ?

File: scripts/Jenkinsfile [hunk 7]
```diff
@@ -47,9 +58,24 @@ pipeline
...
          {
-             junit 'tests/tests_log.xml'
-             echo 'Cleaning Workspace'
-             deleteDir() 
+            script
+            {
+                try 
+                {
+                    sh '''
+                    docker rm -f ${CONTAINER_NAME}
```
Comment (ochougul) [inline scripts/Jenkinsfile:66]: will this conflict when multiple tests are running with same container name?

**Comment by ochougul** [comment]
CI is falling, please fix tests.

**Comment by quic-amitraj** [comment]
Closing because its old version of PR-107.

**Pull request closed without merging**